sequence_processing_pipeline/Pipeline.py

```diff
@@ -144,6 +218,24 @@ def __init__(self, configuration_file_path, run_id, sample_sheet_path,
 
         self.run_dir = self._search_for_run_dir()
 
+        # run_id will be valid assuming there is an existing run-directory
+        # for it.
+        instrument_type = InstrumentUtils.get_instrument_type(run_id)
+
+        if instrument_type in self.configuration['instrument']:
+            # configuration dictionaries for all Jobs have been moved into
+            # an attribute co-level with 'pipeline' called 'instrument'.
+            # Since we only need to know the configuration metadata for the
+            # instrument relevant to this job, extract it, delete the
+            # instrument attribute entirely, and place the extracted metadata
+            # in its legacy position.
+            tmp = self.configuration['instrument'][instrument_type].copy()
```

Comment:
What's happening here is that originally, the resource allocations for bcl2fastq, bcl-convert, QC, etc. were all co-level at the top of the JSON document along with general pipeline configuration metadata. Now there are only two attributes at the top-level: configuration and instrument. Each instrument's resource allocations are stored as a (k,v) pair here.

Rather than change all of the code everywhere else to reflect this new structure, we determine which instrument was used in the run, make a copy of it, delete the entire instrument (k,v) pair, and then move the copied data into its legacy position.